src/Domain/ValueObjects/Wallet.cs

```diff
@@ -4,22 +4,8 @@ namespace Domain.ValueObjects;
 
 public record Wallet(decimal Balance)
 {
-    public User User { get; set; }
-    private decimal _balance = default;
 
-    public decimal Balance
-    {
-        get => _balance;
-        set
-        {
-            if (value < 0)
-            {
-                throw new Exception("number cant be negative");
-            }
-
-            _balance += value;
-        }
-    }
+    public decimal Balance { get; private set; }
 
     public bool HasBalance(decimal amount)
```

Comment:
you can assert (throw ArgumentException if falase) that balance we are checking for is a positive non-zero decimal value. and then just `return Balance >= amount`